src/RewardsStreamerMP.sol

```diff
@@ -99,6 +100,40 @@ contract RewardsStreamerMP is IStakeManager, TrustedCodehashAccess, ReentrancyGu
         account.lastMPUpdateTime = block.timestamp;
     }
 
+    function lock(uint256 lockPeriod) external onlyTrustedCodehash nonReentrant {
+        if (lockPeriod == 0 || (lockPeriod < MIN_LOCKUP_PERIOD || lockPeriod > MAX_LOCKUP_PERIOD)) {
+            revert StakingManager__InvalidLockingPeriod();
+        }
+
+        Account storage account = accounts[msg.sender];
+
+        if (account.lockUntil > 0) {
+            revert StakingManager__AlreadyLocked();
+        }
+
+        if (account.stakedBalance == 0) {
+            revert StakingManager__InsufficientBalance();
+        }
+
+        _updateGlobalState();
+        _updateAccountMP(msg.sender);
+
+        uint256 lockMultiplier = (lockPeriod * MAX_MULTIPLIER * SCALE_FACTOR) / MAX_LOCKUP_PERIOD;
+        uint256 additionalBonusMP = account.stakedBalance * lockMultiplier / SCALE_FACTOR;
```

Comment:
given that we do the same in `stake`, it would be nice if we could extract this to an inline function that based on lockPeriod and amount returns the bonus MP.